backend/src/main/java/gov/cdc/usds/simplereport/db/model/ApiUserRole.java

```diff
@@ -0,0 +1,42 @@
+package gov.cdc.usds.simplereport.db.model;
+
+import gov.cdc.usds.simplereport.config.authorization.OrganizationRole;
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.EnumType;
+import jakarta.persistence.Enumerated;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.ManyToOne;
+import org.hibernate.annotations.JdbcTypeCode;
+import org.hibernate.type.SqlTypes;
+
+@Entity(name = "api_user_role")
+public class ApiUserRole extends AuditedEntity {
+
+  @ManyToOne
+  @JoinColumn(name = "api_user_id", nullable = false)
+  private ApiUser apiUser;
+
+  @ManyToOne
+  @JoinColumn(name = "organization_id", nullable = false)
+  private Organization organization;
+
+  @Column(nullable = false, columnDefinition = "organization_role")
+  @JdbcTypeCode(SqlTypes.NAMED_ENUM)
+  @Enumerated(EnumType.STRING)
+  private OrganizationRole role;
+
+  protected ApiUserRole() {
+    /* for hibernate */
+  }
+
+  public ApiUserRole(ApiUser user, Organization org, OrganizationRole role) {
+    if (role.equals(OrganizationRole.NO_ACCESS)) {
+      throw new IllegalArgumentException(
```

Comment:
I think I'd prefer us to just log this message here for the `NO_ACCESS` edge case and then early return rather than throwing an error. In my mind, doing so will make handling NO_ACCESS easier since by definition, we're saying we're dropping the NO_ACCESS roles and not making them: an early return here would be closer to that definition.

Throwing an error here also makes us need to error handle constructors to these object, which feels like they might cause problems down the road, whereas if we just return void in those cases, we won't have to worry about them.

Reply:
in this case aren't we choosing between the inconvenience of handling the error and the inconvenience of handling a potential empty/invalid ApiUserRole object? between the two I think I'd rather throw an error than create an invalid object. [this answer](https://stackoverflow.com/a/60002733) illustrates the way I'm thinking about this

I agree silently not creating any instances for NO_ACCESS is closer to what we're trying to do, which is what's happening in ApiUser::setRoles, but I think by the time we've called the constructor it may be too late to do that